libs/errors/include/hpx/errors/exception_list.hpp

```diff
@@ -35,7 +34,10 @@ namespace hpx
     class HPX_EXCEPTION_EXPORT exception_list : public hpx::exception
     {
     private:
-        typedef hpx::lcos::local::spinlock mutex_type;
+        // TODO: Does this need to be hpx::lcos::local::spinlock?
+        // typedef hpx::lcos::local::spinlock mutex_type;
+        // TODO: Add correct initialization of boost spinlock.
+        typedef boost::detail::spinlock mutex_type;
```

Comment:
What's the rationale of this change?

Reply:
Not having to depend on `lcos` in this module (with the assumption that we don't actually need HPX thread yielding in the spinlock). But saying that I realize that `exception_list` might live more happily in the `algorithms` module in which case we should be able to keep it as `hpx::lcos::local::spinlock`. What do you think?

Reply:
Is it used only by the algorithms? I'm not sure. Also, assuming we don't need HPX yielding sounds like a bit of a stretch... However, I see your point...

Reply:
Hrm, almost only by algorithms (and executors)... two other files include it.

I know no HPX thread yielding does sound a bit like a stretch, but I don't see it as a performance problem since we're dealing with exceptions anyway in which case something has already gone wrong. That leaves the question whether this can deadlock. I can't prove this won't happen but I'm hoping our tests are extensive enough to tell us.

Having `lcos` as a dependency would make modularization very difficult so I'd rather try to make it work with the boost spinlock before resorting to that.

Reply:
Ok, fair enough. BTW, there is also `hpx::util::spinlock` that relies on Boost's version but additionally does lock-tracking for HPX. So using this instead might actually be the better option.

Reply:
Agreed, but `hpx::util::spinlock` depends on exceptions making that problematic as well. This should one of, if not the, only module that would have to do this (depend directly on `boost::spinlock`). `hpx::util::spinlock` is coming up right after this module.